### ISSUE TITLE
Error handling in `Database.query` broken as of 2.4.0

### ISSUE THREAD

**Issue by cuzzlor** [ibmdb/node-ibm_db#415]
Since 2.4.0 we find that if we issue queries returning errors using the async (Promise returning) version of the `Database.query` method, the promise is never resolved / the error isn't raised.

If I roll back our reference to 2.3.1 it works as expected.

```ts
        try {
            return await db.query('select 1 from sysibm.syszzzz where 1=1');
        } catch (e) {
            // on v2.4.0 this catch is never executed and nothing is returned
            console.log(e.message, e.stack);
        }
```

**Comment by huineng** [comment]
i agree  , this is a serious issue also using pool,   fact is that because no error is returned we get a timout

**Comment by huineng** [comment]
I used your test case promise-test-open-close as a start

```
var common = require("./common")
  , odbc = require("../")
  , db = new odbc.Database()
  , assert = require("assert");

//assert.equal(db.connected, false);

db.query("select * from " + common.tableName).then(null, function (err) {
  assert.deepEqual(err, { message: 'Connection not open.' });
  assert.equal(db.connected, false);
});

db.open(common.connectionString)
  .then(function (res) {
    assert.equal(db.connected, true);

    db.query("select * from " + common.tableName)
      .then(null, function (err) {
        console.log(err);
        assert.deepEqual(err, { message: 'Connection not open.' });
        assert.equal(db.connected, false);
      });

  });
```

i ran it agains  2.4.0 
```
node promise-test-open-close

```
and " 2.3.9

```
node promise-test-open-close
{ Error: [IBM][CLI Driver][DB2/LINUXX8664] SQL0204N  "QFIND.NODE_ODBC_TEST_TABLE" is an undefined na
me.  SQLSTATE=42704

  errors: [],  error: '[node-ibm_db] SQL_ERROR',
  message: '[IBM][CLI Driver][DB2/LINUXX8664] SQL0204N  "QFIND.NODE_ODBC_TEST_TABLE" is an undefined
 name.  SQLSTATE=42704\n',
  state: '42S02' }
```

2.4.0 didn't produce any error and stopped

in my production system this leads to a timeout 

hope this helps

**Comment by cuzzlor** [comment]
Hi @bimalkjha , any chance we can see a 2.4.1 release with this fix in? We are stuck with a fixed dependency on 2.3.1 currently.

**Comment by huineng** [comment]
Question .. why are you closing this ?  the issue is not fixed, there's no published package that contains you fix

Installing directly the github repo is not working either as this creates other problems) 

thanks

for me this is a bug fix and i would suggest publishing more often bug releases

thanks
